### PR TITLE
add auto_attach param to katello_activation_key

### PR DESCRIPTION
**Pull request by sean797**

### CODE REVIEW

**Comment by mdellweg** [comment]
Mind to add a test case, while you are at it?

File: modules/katello_activation_key.py [hunk 1]
```diff
@@ -144,19 +150,23 @@ def update_fields(self, new, old, fields):
             if hasattr(new, field) and hasattr(old, field):
                 new_attr = getattr(new, field)
                 old_attr = getattr(old, field)
-                if old_attr is None or new_attr.id != old_attr.id:
+                if (
+                   (old_attr is None)
+                   or (hasattr(new_attr, 'id') and hasattr(old_attr, 'id') and new_attr.id != old_attr.id)
+                   or (not hasattr(new_attr, 'id') and not hasattr(old_attr, 'id') and new_attr != old_attr)
+                   ):
```
Comment (sean797) [inline modules/katello_activation_key.py]: Added `--ignore W503`, personally I think this is as easy to read, though suggestions welcome

Comment (mdellweg) [reply modules/katello_activation_key.py]: Not sure i like it...

Comment (mdellweg) [reply modules/katello_activation_key.py]: This looks a lot like
https://github.com/theforeman/foreman-ansible-modules/blob/eff95ed281e7e50abf1c5024bb7709e26332f34d/module_utils/ansible_nailgun_cement.py#L215
But i'd prefer using
https://github.com/theforeman/foreman-ansible-modules/blob/eff95ed281e7e50abf1c5024bb7709e26332f34d/module_utils/ansible_nailgun_cement.py#L292
a i think it is more specific in deciding whether a value is an entity.
Ideally, you could use naildown_entity_state. We might get deleting for free. ;)

Comment (sean797) [reply modules/katello_activation_key.py]: ohh nice, updated to use `fields_equal`

**Pull request merged**